File: sellergoods-service/src/main/java/com/hapiio/sellergoodsservice/service/BrandService.java
package com.hapiio.sellergoodsservice.service;

import com.hapiio.commons.common.PageResult;
import com.hapiio.commons.common.Result;
import com.hapiio.pojo.model.Brand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    List<Map> selectOptionList();

    List<Brand> finAll();

    Result batch(List<Brand> brands);

    Result addBrand(Brand brand);

    Result updateBrand(Brand brand);

    PageResult findPage(Integer pageNo, Integer pageSize);

    Brand findBrand(Long id);

    Result deleteBrand(Long[] ids);

    PageResult search(Integer pageNo, Integer pageSize, Brand brand);
}

File: manager-web/src/main/resources/static/js/Service/brandService.js
app.service('brandService',function ($http) {

    //读取列表数据绑定到表单中

    this.selectOptionList=function(){
        return $http.get('/admin/brand/selectOptionList');
    }

    this.findAll = function(){
        return $http.get("/admin/brand/findAll");
    }

    this.findPage = function(pageNo,pageSize){
        return $http.get("/admin/brand/findPage?pageNo="+pageNo+"&pageSize="+pageSize);
    }

    this.update = function(entity){
        return $http.post("/admin/brand/update",entity);
    }

    this.add = function(entity) {
        return $http.post("/admin/brand/add",entity);
    }

    this.findOne = function(id){
        return $http.get("/admin/brand/findOne?id="+id);
    }

    this.search = function(pageNo,pageSize,entity){
        return $http.post("/admin/brand/search?pageNo="+pageNo+"&pageSize="+pageSize,entity);

    }

    this.delete = function(ids){
        return $http.get("/admin/brand/delete?ids="+ids);
    }


})